src/include/server/mir/frontend/keyboard_helper.h

```diff
@@ -74,6 +73,15 @@ class KeyboardHelper
     void refresh_modifiers();
 
 private:
+    friend class mir::frontend::WlSeat;
+    friend class mir::shell::AccessibilityManager;
+    KeyboardHelper(
+        KeyboardCallbacks* keybaord_impl,
+        std::shared_ptr<mir::input::Keymap> const& initial_keymap,
+        std::shared_ptr<input::Seat> const& seat,
+        int default_repeat_rate,
+        int default_repeat_delay);
+
```

Comment:
Why are we making the constructor private? And adding `friends`?

Especially making `mir::shell::AccessibilityManager` a friend looks wrong: why would `shell` code know anything about the `frontend` implementation classes?